src/panels/config/entities/ha-config-entities.ts

```diff
@@ -156,18 +167,38 @@ export class HaConfigEntities extends SubscribeMixin(LitElement) {
   private _filteredEntities = memoize(
     (
       entities: EntityRegistryEntry[],
+      states: HassEntities,
       showDisabled: boolean,
-      showUnavailable: boolean
+      showUnavailable: boolean,
+      showReadOnly: boolean
     ) => {
+      let stateEntities: EntityRegistryEntry[] = [];
+      if (showReadOnly) {
+        const regEntityIds = entities.map((entity) => entity.entity_id);
+        const stateEntityIds = Object.keys(states).filter(
+          (entityId) => !regEntityIds.includes(entityId)
+        );
+        stateEntities = stateEntityIds.map((entityId) => {
+          return {
+            name: computeStateName(states[entityId]),
+            entity_id: entityId,
+            platform: computeDomain(entityId),
+            disabled_by: null,
+            readonly: true,
+            selectable: false,
+          };
+        });
+      }
+
       if (!showDisabled) {
         entities = entities.filter((entity) => !Boolean(entity.disabled_by));
       }
 
-      return entities.reduce((result, entry) => {
-        const state = this.hass!.states[entry.entity_id];
+      return entities.concat(stateEntities).reduce((result, entry) => {
```

Comment:
I know that this is existing code, but `reduce` creates less readable code. Much prefer `for… of` combined with `continue`.